components/automate-ui/src/app/page-components/services-sidebar/services-sidebar.component.spec.ts

```diff
@@ -0,0 +1,232 @@
+    store = TestBed.inject(Store);
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  describe('when selecting and deselecting checkboxes', () => {
+    beforeEach(() => {
+      mockServices = [
+      {
+        id: 300,
+        supervisor_id: '1af6ecdc-acdd-421c-94f7-ec4f8c128691',
+        release: 'custom/nginx/1.0.1/20190115184823',
+        group: 'nginx.default',
+        status: 'ok',
+        health_check: 'CRITICAL',
+        applications: 'bldr-cache',
+        environment: 'qa',
+        fqdn: '8ouxxga6.example.net',
+        site: 'testsite',
+        channel: 'stable',
+        update_strategy: 'AT-ONCE',
+        previous_health_check: 'NONE',
+        current_health_since: '23 hours 54 minutes',
+        health_updated_at: new Date(),
+        health_check_result: {
+            stdout: 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF…FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF', stderr: 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF…FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF', exit_status: 2
+          }
+        },
+        {
+          id: 400,
+          supervisor_id: '1af6ecdc-acdd-421c-94f7-ec4f8c128691',
+          release: 'custom/nginx/1.0.1/20190115184823',
+          group: 'nginx.default',
+          status: 'ok',
+          health_check: 'CRITICAL',
+          applications: 'bldr-cache',
+          environment: 'qa',
+          fqdn: '8ouxxga6.example.net',
+          site: 'testsite',
+          channel: 'stable',
+          update_strategy: 'AT-ONCE',
+          previous_health_check: 'NONE',
+          current_health_since: '23 hours 54 minutes',
+          health_updated_at: new Date(),
+          health_check_result: {
+            stdout: 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF…FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF', stderr: 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF…FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF', exit_status: 2
+          }
+        },
+        {
+          id: 500,
+          supervisor_id: '1af6ecdc-acdd-421c-94f7-ec4f8c128691',
+          release: 'custom/nginx/1.0.1/20190115184823',
+          group: 'nginx.default',
+          status: 'ok',
+          health_check: 'CRITICAL',
+          applications: 'bldr-cache',
+          environment: 'qa',
+          fqdn: '8ouxxga6.example.net',
+          site: 'testsite',
+          channel: 'stable',
+          update_strategy: 'AT-ONCE',
+          previous_health_check: 'NONE',
+          current_health_since: '23 hours 54 minutes',
+          health_updated_at: new Date(),
+          health_check_result: {
+            stdout: 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF…FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF', stderr: 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF…FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF', exit_status: 2
+          }
+        }
+      ];
+
+      component.servicesList = mockServices;
+    });
+
```

Comment:
was having difficulty grabbing elements, so most of these test functions instead of behavior.